completions/ssh.completion.sh

```diff
@@ -18,6 +20,17 @@ function _omb_completion_ssh {
     COMPREPLY=($(compgen -W "$(grep ^Host "$HOME/.ssh/config" | awk '{for (i=2; i<=NF; i++) print $i}' )" "${options[@]}"))
   fi
 
+  # check if .ssh/config contains Include options
+  for additional_include_option in $(awk -F' ' '/^Include/{print $2}' "$HOME/.ssh/config" 2>/dev/null) ;do
+    # relative or absolute path, if relative transforms to absolute
+    [[ "${additional_include_option:0:1}" == "/" ]] ||additional_include_option="$HOME/.ssh/$additional_include_option"
+    # for loop to interpret possible globing
+    for additional_include_defined_file in $additional_include_option ;do
+      # parse all defined hosts from that file
+      [[ -s "$additional_include_defined_file" ]] &&COMPREPLY+=($(compgen -W "$(grep ^Host "$additional_include_defined_file" | awk '{for (i=2; i<=NF; i++) print $i}' )" "${options[@]}"))
+    done
```

Comment:
We are now performing the command substitutions for every included file, but I think we can first collect the valid file names in an array, and then perform the search at once by specifying multiple files to awk as `awk '/^Host /{for (i=2; i<=NF; i++) print $i}' "${array[@]}"` (I think there is no strong reason to separate the processing in `grep` and `awk`. We can perform both in `awk`) (Note: the actual array name should be chosen properly instead of just `array`). This reduces the fork/exec cost. What do you think?

Reply:
I think that I'll certainly learn a few things trying to improve that code :-p
I'll look at it!

Reply:
> We are now performing the command substitutions for every included file, but I think we can first collect the valid file names in an array, and then perform the search at once by specifying multiple files to awk

6c221ff

> (Note: I also think there is no strong reason to separate the processing in `grep` and `awk`. We can perform both in `awk`)

a55af5b